bitcask.go

```
@@ -273,8 +259,47 @@ func (b *Bitcask) Fold(f func(key []byte) error) (err error) {
 	return
 }
 
+// get retrieves the value of the given key. If the key is not found or an/I/O
+// error occurs a null byte slice is returned along with the error.
+func (b *Bitcask) get(key []byte) (internal.Entry, error) {
+	var df data.Datafile
+
+	b.mu.RLock()
+	value, found := b.trie.Search(key)
+	if !found {
+		b.mu.RUnlock()
+		return internal.Entry{}, ErrKeyNotFound
+	}
+
+	item := value.(internal.Item)
+
+	if item.FileID == b.curr.FileID() {
+		df = b.curr
+	} else {
+		df = b.datafiles[item.FileID]
+	}
+
+	e, err := df.ReadAt(item.Offset, item.Size)
+	b.mu.RUnlock()
+	if err != nil {
+		return internal.Entry{}, err
+	}
+
+	if e.Expiry != nil && e.Expiry.Before(time.Now().UTC()) {
+		go b.Delete(key)
```

Why is this a goroutine? We're not checking for errors here. Should we?

---

Deletion of key can be done lazily, ie. client response does not require to be put on hold. We can ignore err here because once a key gets expired, it is expired unless re-written so each time client will get `ErrKeyExpired`. Do you have some suggestions on this?